### PR TITLE
Various pypy3Packages fixes

### PR DESCRIPTION
**Pull request by mweinelt**
Allows building up to `pypy3Packages.cryptography`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/virtualenv/default.nix [hunk 1]
```diff
@@ -76,6 +75,10 @@ buildPythonPackage rec {
     "test_seed_link_via_app_data"
     # Permission Error
     "test_bad_exe_py_info_no_raise"
+  ] ++ lib.optionals (isPyPy) [
+     "test_bash"
+     "test_can_build_c_extensions"
+     "test_discover_ok"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/virtualenv/default.nix:81]: Those just don't work on pypy?

File: pkgs/development/python-modules/yapf/default.nix [hunk 2]
```diff
@@ -13,6 +14,8 @@ buildPythonPackage rec {
     hash = "sha256-o/UIXTfvfj4ATEup+bPkDFT/GQHNER8FFFrjE6fGfRs=";
   };
 
+  doCheck = !isPyPy;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yapf/default.nix:17]: This should also get a comment

Comment (SuperSandro2000) [reply pkgs/development/python-modules/yapf/default.nix:17]:
```suggestion
  # uses nosetests which is not available on pypy
  doCheck = !isPyPy;
```

File: pkgs/development/python-modules/virtualenv/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/virtualenv/default.nix:81]:
```
virtualenv> FAILED tests/unit/activation/test_bash.py::test_bash[with_prompt] - AssertionError: /nix/store/hd9sv84a4b5yk3kvnpgy51qqd4zh8bf5-pypy3.9-7.3.11/...
virtualenv> FAILED tests/unit/activation/test_bash.py::test_bash[no_prompt] - AssertionError: /nix/store/hd9sv84a4b5yk3kvnpgy51qqd4zh8bf5-pypy3.9-7.3.11/...
virtualenv> FAILED tests/unit/create/via_global_ref/test_build_c_ext.py::test_can_build_c_extensions[venv] - PermissionError: [Errno 13] Permission denied: PosixPath('/build/pytest-of-...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3.9.16-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3.9.16--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3.9-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3.9--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[PyPy-3--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3.9.16-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3.9.16--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3.9-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3.9--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3-64-bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...
virtualenv> FAILED tests/unit/discovery/py_info/test_py_info_exe_based_of.py::test_discover_ok[python-3--bin-] - RuntimeError: failed to detect pypy3.9.16-64|pypy3.9.16|pypy3.9-64|pypy3.9|...

File: pkgs/development/python-modules/yapf/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/yapf/default.nix:18]: Added.

File: pkgs/development/python-modules/virtualenv/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/virtualenv/default.nix]: Added comments.

**Pull request merged**